### PR TITLE
Dependency Updates to Support new version of Angular/TSLint

### PR DESCRIPTION
**Pull request by tzfx**
It looks like there were some dependency requirements that needed to be updated that were getting masked by our package-lock.json.

I've updated these dependencies, and included some peer deps that also required inclusion or upgrade.

Additionally, it looks like the karma config is attempting to report coverage statistics by default? I've added the `--code-coverage` flag to the `npm run test` script so this works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,21 +22,24 @@
     "@angular/compiler": "9.1.0",
     "@angular/core": "9.1.0",
     "@angular/forms": "9.1.0",
+    "@angular/localize": "9.1.0",
     "@angular/platform-browser": "9.1.0",
     "@angular/platform-browser-dynamic": "9.1.0",
     "@angular/router": "9.1.0",
-    "@ng-select/ng-select": "3.7.3",
+    "@ng-select/ng-select": "^4.0",
```
Comment (tzfx) [inline package.json:29]: This was the one that made me start looking at our dependencies-- ng-select 4.x is 9 compatible, 3.x is 8 compatible.

File: package.json [hunk 2]
```diff
@@ -46,7 +49,7 @@
     "@types/jasmine": "3.5.0",
     "@types/jasminewd2": "2.0.3",
     "@types/node": "^12.11.1",
-    "codelyzer": "^5.1.2",
+    "codelyzer": "^5.2.2",
```
Comment (tzfx) [inline package.json:52]: 5.1 had a dep on tslint 5.

File: package.json [hunk 3]
```diff
@@ -68,7 +71,7 @@
     "ts-node": "8.3.0",
     "tslint": "6.1.0",
     "tslint-config-prettier": "^1.18.0",
-    "tslint-plugin-prettier": "^2.1.0",
+    "tslint-plugin-prettier": "^2.3.0",
```
Comment (tzfx) [inline package.json:74]: 2.1 had a dep on tslint 5.

File: package.json [hunk 4]
```diff
@@ -22,21 +22,24 @@
...
     "@angular/platform-browser": "9.1.0",
     "@angular/platform-browser-dynamic": "9.1.0",
     "@angular/router": "9.1.0",
-    "@ng-select/ng-select": "3.7.3",
+    "@ng-select/ng-select": "^4.0",
     "@types/socket.io-client": "^1.4.32",
     "bootstrap": "4.3.1",
     "core-js": "2.5",
     "font-awesome": "4.7",
+    "jquery": "^3.4.1",
```
Comment (tzfx) [inline package.json:34]: Jquery/popper are peers for angular stuff.

File: package.json [hunk 5]
```diff
@@ -22,21 +22,24 @@
     "@angular/compiler": "9.1.0",
     "@angular/core": "9.1.0",
     "@angular/forms": "9.1.0",
+    "@angular/localize": "9.1.0",
```
Comment (tzfx) [inline package.json:25]: The updated ng-select has a peer dep on localize.

File: package.json [hunk 6]
```diff
@@ -8,8 +8,8 @@
     "start:prod": "ng serve --proxy-config proxy.conf.json --prod",
     "build": "npm run lint && ng build",
     "build:prod": "npm run lint && ng build --prod",
-    "build:bundle-report": "ng build --stats-json && webpack-bundle-analyzer dist/ngx-starter/stats-es5.json",
-    "test": "ng test",
+    "build:bundle-report": "ng build --stats-json && webpack-bundle-analyzer dist/ngx-starter/stats-es2015.json",
```
Comment (tzfx) [inline package.json:11]: Our default target is now es2015-- this made the bundle-report command very unhappy.

File: package.json (hunk 1 above)
Comment (jrassa) [reply package.json:29]: Looks like they just released 4.0.  It wasn't available when I did the upgrade.  There were build warnings, but it still seemed to work properly.

**Pull request merged**